463 - added previous day and next day arrows on mobile view

azjy89 opened the pull request:
also added some tests however they are not very extensive no comprehensive and may need to be improved upon.

JessicaF commented on frontend/__tests__/BookingCalendar.test.tsx:
```diff
@@ -0,0 +1,110 @@
+
+describe("Booking Calendar", () => {
+  beforeAll(() => {
+    // Suppress console.warn messages
```
Why are we supressing console.warn message?

JessicaF commented on frontend/__tests__/BookingCalendar.test.tsx:
```diff
@@ -0,0 +1,110 @@
+import "@testing-library/jest-dom";
+
+import { Booking } from "@common/types";
+import NavigateNextIcon from "@mui/icons-material/NavigateNext";
```
Can you please get rid of all unused imports?

JessicaF commented on frontend/__tests__/BookingCalendar.test.tsx:
```diff
@@ -0,0 +1,110 @@
+  const end: Date = new Date();
+  const events: Booking[] = [
+    {
+      name: "s",
```
Nit but would be better if the mock data resembled the real content better

JessicaF commented on frontend/__tests__/BookingCalendar.test.tsx:
```diff
@@ -0,0 +1,110 @@
+    expect(heading).toBeInTheDocument();
+  });
+
+  it("renders the outer stack", () => {
```
Instead of testing if a `Stack` is rendered, test the things inside of this element that users can see/interact with

JessicaF commented on frontend/__tests__/BookingCalendar.test.tsx:
```diff
@@ -0,0 +1,110 @@
+      </Provider>
+    );
+
+    const calendarContainer = queryByTestId("calendarContainer");
```
Prefer not to test if a container box is rendered but rather its contents and what the user will interact with

JessicaF commented on frontend/components/BookingCalendar.tsx:
```diff
@@ -238,29 +242,66 @@ const BookingCalendar: React.FC<{ events: Array<Booking> }> = ({ events }) => {
         width="100%"
         // px={{ xs: 3, md: 15 }}
         pt={3}
+        data-testid="outerStack"
```
Using `data-testid` is usually a last-resort - prefer to use the [queries offered by RTL](https://testing-library.com/docs/queries/about#priority)

JessicaF commented on frontend/components/BookingCalendar.tsx:
```diff
@@ -238,29 +242,66 @@ const BookingCalendar: React.FC<{ events: Array<Booking> }> = ({ events }) => {
+              data-testid="dateSelectStack"
+            >
+              {isMobile && (
+                <IconButton
```
The `IconButton` could probably be put into a different component file and imported into here instead. This will reduce the repetition and verbosity of the code.

JessicaF reviewed (changes requested):
@azjy89 I've left a few comments in the code for you to take a look at - I didn't comment on every single thing you need to change but just the first instance of them instead. Lemme know if you have ay questions!

Otherwise functionality is looking good! 

One minor aesthetic change to the hover state of the arrows would be nice (i.e. can we make it a box instead of circle so it matches the aesthetic more and also make it so the hover has a gap with the date box):  
<img width="739" alt="Screenshot 2024-04-05 at 11 58 23 AM" src="https://github.com/devsoc-unsw/freerooms/assets/54833118/16054a2f-1b3d-46fc-b860-e17bb86fdf58">

azjy89 commented:
In my new commits i have fixed the console warn issue
![image](https://github.com/devsoc-unsw/freerooms/assets/145690917/af89d867-b8c8-4c36-b84a-5cdf344f5713)
by turning on disableMaskedInput for the MUI date picker component. 

Also rather than testing the containers I have tested the components which are seen by the viewer (namely the buttons and the Room Bookings)
![image](https://github.com/devsoc-unsw/freerooms/assets/145690917/cc8060e3-27d5-4662-a836-d4422b90b24c)
as well as the calendar as a whole. 

I changed to the best possible RTL queries using testing playground however some still can only use data-testid. 

Mock data has been updated to better reflect real use.

Unnecessary imports have been removed.

The hover effect has been removed from the icon buttons to fit in better with the style of the rest of the page

JessicaF commented:
Thank you for the changes !! 😎 I'll try see if I can find a way around the data-testid stuff/IconButton component (not sure if I can haha) but otherwise we'll merge you code in after I give it a go haha

JessicaF commented:
Hi @azjy89, I've made some changes to your code - take a look and lmk what you think! I've updated `BookingCalendar` to use Date Picker v6 instead of v5 and combined it with other changes from the `dev` branch. I also changed the tests, in particular for checking if the correct icon buttons are shown in mobile view.

azjy89 commented:
Hi @JessicaF, I had a look at the made changes I think it looks good, everything works as intended, and everything makes sense to me now about the testing. Just about the disableRipple (below) on the IconButton do you want the disableRipple enabled to remove the circular onClick effect on the icon button or to leave the effect there?
![image](https://github.com/devsoc-unsw/freerooms/assets/145690917/2d1d1fe9-81b1-4110-9d95-d749459cc4b7)

This is what it looks like disabled (below)
![image](https://github.com/devsoc-unsw/freerooms/assets/145690917/665e8313-53a6-49eb-b9ce-943d095a39b8)
If this I will just merge.

and this is what it looks like enabled (no effect)
![image](https://github.com/devsoc-unsw/freerooms/assets/145690917/972d18b5-e4ef-4a09-93ee-458938ef06c0)
If this I'll make this change and then commit. 

Thanks for reviewing 👍

JessicaF commented:
Hi @azjy89, I'm fine with having the ripple (i.e. not adding `disableRipple`) because on my side the buttons are no long circular but more square (fixed through having the `borderRadius: 3`: 

<img width="432" alt="Screenshot 2024-05-21 at 2 21 10 PM" src="https://github.com/devsoc-unsw/freerooms/assets/54833118/30ae2a33-bfb6-4f54-b8b5-c0141728ffe1">

I think it's fine to just merge in my final commit as long as everything else is okay!
